client/src/lib/dependency-injection/FormStateManager.js

```diff
@@ -19,25 +35,25 @@ class SchemaStateManager {
    * @return {object}
    */
   getFieldByName(fieldName) {
-    return this.schemaState.fields.find(field => field.name === fieldName);
+    return this.schema.state.fields.find(field => field.name === fieldName);
   }
 
   /**
    * Updates a field by callback
    * @param {string} fieldName
    * @param {function} updater
-   * @returns {SchemaStateManager}
+   * @returns {FormStateManager}
    */
   mutateField(fieldName, updater) {
-    const fieldList = this.schemaState.fields || [];
+    const fieldList = this.schema.state.fields || [];
     const fieldIndex = fieldList.findIndex(field => field.name === fieldName);
     if (fieldIndex < 0) {
       return this;
     }
     const field = fieldList[fieldIndex];
```

Comment:
just noticed this should be `{ ...fieldList[fieldIndex] }` or below have `updater({ ...field })` in case of mutations in the `updater()`